app/assets/javascripts/discourse/app/components/load-more-accessible.gjs

```diff
@@ -0,0 +1,271 @@
+  }
+
+  get buttonAriaLabel() {
+    return this.isLoadingAbove
+      ? i18n("post.sr_load_more_posts_above")
+      : i18n("post.sr_load_more_posts_below");
+  }
+
+  @action
+  async handleIntersectionLoad() {
+    return this.#loadWithFocusManagement();
+  }
+
+  async #loadWithFocusManagement() {
+    if (!this.enabled || !this.canLoadMore || this.isLoading) {
+      return;
+    }
+
+    try {
+      this.isLoading = true;
+
+      // Announce loading has started
+      this.#announceLoading();
+
+      // Get existing post numbers and current focus context before loading
+      const existingPostNumbers = this.#getExistingPostNumbers();
+      const currentFocusContext =
+        this.#getCurrentFocusContext(existingPostNumbers);
+
+      // Trigger the actual loading action
+      await this.args.action();
+
+      // After loading completes, move focus to the appropriate new post
+      schedule("afterRender", () => {
+        setTimeout(() => {
+          this.#focusAppropriateNewPost(
+            existingPostNumbers,
+            currentFocusContext
+          );
+        }, 300);
+      });
+    } finally {
+      this.isLoading = false;
+    }
+  }
+
+  #getCurrentFocusContext(existingPostNumbers) {
+    if (this.isLoadingAbove) {
+      // For loading above, find all posts in DOM order and get the first one
+      // (since new posts will be inserted above it)
+      const allPostElements = Array.from(
+        document.querySelectorAll("[data-post-number]")
+      );
+      const firstPostNumber =
+        allPostElements.length > 0
+          ? parseInt(allPostElements[0].dataset.postNumber, 10)
+          : Math.min(...existingPostNumbers);
+
+      return {
+        direction: "above",
+        nearestPost: firstPostNumber,
+        allExistingInOrder: allPostElements.map((el) =>
+          parseInt(el.dataset.postNumber, 10)
+        ),
+      };
+    } else {
+      // For loading below, find all posts in DOM order and get the last one
+      // (since new posts will be inserted below it)
+      const allPostElements = Array.from(
+        document.querySelectorAll("[data-post-number]")
+      );
+      const lastPostNumber =
+        allPostElements.length > 0
+          ? parseInt(
+              allPostElements[allPostElements.length - 1].dataset.postNumber,
+              10
+            )
+          : Math.max(...existingPostNumbers);
+
+      return {
+        direction: "below",
+        nearestPost: lastPostNumber,
+        allExistingInOrder: allPostElements.map((el) =>
+          parseInt(el.dataset.postNumber, 10)
+        ),
+      };
+    }
+  }
+
+  #announceLoading() {
+    const announcement = document.createElement("div");
+    announcement.setAttribute("aria-live", "assertive");
+    announcement.className = "sr-only";
+    announcement.textContent = this.isLoadingAbove
+      ? "Loading more posts above..."
+      : "Loading more posts below...";
+    document.body.appendChild(announcement);
```

Comment:
Ideally, we should avoid creating DOM elements directly like this. 

Does the component have to be added to the `body` element? If it does we could create an announcement component and a service to provide this capability to other components.

Otherwise, we can render it below in the template.

In case it does.